doorpi/action/SingleActions/take_snapshot.py

```diff
@@ -0,0 +1,62 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+import logging
+import os
+logger = logging.getLogger(__name__)
+logger.debug("%s loaded", __name__)
+
+SIPPHONE_SECTION = 'SIP-Phone'
+DOORPI_SECTION = 'DoorPi'
+
+from action.base import SingleAction
+import doorpi
+import subprocess as sub
+
+conf = doorpi.DoorPi().config
+
+def take_snapshot(size, path, max):
+    if not os.path.exists(path):
+        logger.info('Path (%s) does not exist - creating it now', path)
+        os.makedirs(path)
+
+    lastFile = getLastFilename(path)
+    lastNr = 1
+    if (len(lastFile) > 0):
+        lastNr = int(lastFile[:lastFile.rfind(".jpg")])
+        if (lastNr+1 <= max):
+            lastNr = lastNr + 1
+	else:
+	    lastNr = 1
+    imageFilename = path + str(lastNr) + ".jpg"
+    # fswebcam automatically selects the first video device
+    command = "fswebcam --top-banner -b --font luxisr:20 -r " + size + " " + imageFilename
+    p = sub.Popen(command, shell=True, stdout=sub.PIPE, stderr=sub.PIPE)
+    output, errors = p.communicate()
+    if (len(errors) > 0):
+        logger.error('error creating snapshot - maybe fswebcam is missing')
+    else:
+        logger.info('snapshot created: %s', imageFilename)
+    return
+
+
+def getLastFilename(path):
```

Comment:
Hinweis auf http://code.activestate.com/recipes/576804-find-the-oldest-or-yougest-of-a-list-of-files/#c1